tests/test_kubernetes_system.py

```diff
@@ -0,0 +1,53 @@
+# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
+# Copyright (c) 2024 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from unittest.mock import MagicMock, patch
+
+import pytest
+from cloudai.systems.kubernetes.kubernetes_system import KubernetesSystem
+from kubernetes import client
+
+
+@pytest.fixture
+def k8s_system():
+    """Fixture to create a KubernetesSystem instance."""
+    with patch("kubernetes.config.load_kube_config"), patch("pathlib.Path.exists", return_value=True):
+        k8s_system = KubernetesSystem(
+            name="test-system",
+            install_path=Path("/fake/install/path"),
+            output_path=Path("/fake/output/path"),
+            kube_config_path=Path("/fake/kube/config"),
+            default_namespace="default",
+            default_image="test-image",
+        )
+        k8s_system._core_v1 = MagicMock(client.CoreV1Api)
+        k8s_system._batch_v1 = MagicMock(client.BatchV1Api)
+        k8s_system._custom_objects_api = MagicMock(client.CustomObjectsApi)
+        yield k8s_system
+
+
+def test_initialization(k8s_system):
```

Comment:
I'm not sure what this test actually verifies... It just mock everything, not that much code under test. I'd focus on:
1. `kube_config_path` existence.
2. `core_v1` and similar initialization.

For all that I suggest using raw dicts and `KubernetesSystem.model_validate(...)` method to have better coverage.

Reply:
I wanted to test the initialization of member variables because it failed in the main branch.

> kube_config_path existence.

We cannot test this because the configuration file does not exist in the pytest environment.

> core_v1 and similar initialization.

I intentionally did not test the initialization of these for two reasons. First, I am working on refactoring these variables, as shown in https://github.com/NVIDIA/cloudai/pull/197. Second, I am unsure whether the initialization will succeed because the configuration file is missing anyway.

Reply:
@amaslenn  Let's remove these unit tests if your initial intention was to test the API member variables. I assumed you wanted to verify whether the other member variables were initialized properly.

Reply:
Updated, @amaslenn

Reply:
I see that use don't use `KubernetesSystem.model_validate(...)`, could you please explain why? IMO that would be a better reproducer.

Reply:
I missed it. Updated it.